proxy.js

```diff
@@ -0,0 +1,142 @@
+/*
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESSED OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+'use strict';
+
+var fs = require('fs');
+var _ = require('lodash');
+var traverse = require('traverse');
+
+
+
+/**
+ * proxy injector
+ *
+ * for each target determine if proxies are enabled. If this is the case inject proxy containers into 
+ * each machine. A machine is defined as a container that holds child docker or process containers.
+ * This is done pre-compilation to the objects in the topology tree. In addition a proxy definition is 
+ * added to the list of container definitions
+ */
+module.exports = function() {
+
+  /**
+   * nscale proxy definition, this will execute a container that holds the 
+   */
+  var proxyCDef = {
+    type: 'docker',
+    specific: {
+      repositoryUrl: 'https://github.com/nearform/nscale-proxy-container.git',
```

Comment:
Is this really needed? Could we just use the 'stock' haproxy image from Docker Hub, and then customize it via `docker.exec`?